tests/README.md

```diff
@@ -0,0 +1,46 @@
+# Rancher Test Framework
+
+The Rancher Test Framework provides tools for writing integrations and validation tests.  The framework manages interactions with the external services being tested and aids in cleaning up resources after a test is completed.  The framework is organized into three disciplines: framework, clients, and extensions.  The framework consists of a few core libraries used to make it easy to write homologous tests. See [extensions](#extensions) and [clients](#clients) for more details.
+
+## Requirements
+
+---
+
+#### Integration
+To run rancher integration tests you will need:
+- a running instance of rancher with an accessible url
+- a rancher access token for the admin user
+- [golang 1.16](https://go.dev/doc/install)
+- [k3d](https://k3d.io/v5.1.0/)
+
+#### Validation
+Validation tests have the same requirements as integration tests however different suites may need credentials for cloud providers.  Check the configs for your test suite for details.
+
+## Concepts
+
+---
+
+### Integration vs Validation
+
+TODO
+
+### Extensions
+
+Extensions are functions that complete common operations used by tests.  Extensions should not require much configuration or support multiple behaviors.  Extensions should be simple enough that they can be used in many tests without needing to be tested themselves.
+
+### Clients
+
+TODO
+
+### Sessions
+
+Sessions are used to track resources created by tests.  A session allows cleanup functions to be registered while it is open.  Once a session is closed the cleanup functions will be called latest to oldest.  Sessions should be closed after a set of tests that use the same resources is completed.  This eliminates the need for each test to create and tear down its own resources allowing for more efficient reuse of some resources.
+
```

Comment:
Maybe we should also emphasize not needing to keep track of each resource created, and then clean them up individually. That Sessions will keep track of all of that. Just a thought